src/main/java/org/plumelib/util/ArraysPlume.java

```diff
@@ -4121,15 +4121,22 @@ boolean isPartitioningFor(List<T> elts) {
      * @param elt the element to add
      * @return a new partitioning just like this one, but with elt added to the ith part
      */
+    @SuppressWarnings("index:argument")
     Partitioning<T> addToPart(@NonNegative int i, T elt) {
       Partitioning<T> result = new Partitioning<>(this);
-      if (size() == i) {
+      int n = result.size();
+
+      if (i == n) {
+        // append a new part at the high bound (i== size() is allowed)
         ArrayList<T> newPart = newArrayList(elt);
         result.add(newPart);
-      } else {
+      } else if (0 <= i && i < n) {
+        // explicit lower/upper bounds let the Index Checker prove get/set are in-bounds
```

Comment:
That's true, but a type-checker should not be require code changes.  What would it take to eliminate the warning without code changes?